docs/template-strings.md

```diff
@@ -1,62 +1,64 @@
+### Şablon Stringler  
+Tek tırnak veya çift tırnak ile tanımlanan stringler yerine aksan işareti (\`) ile tanımlanmış stringlerdir. Şablon Stringler aşağıdaki amaçlar için kullanılır:
 
-* String Interpolation
-* Multiline Strings
-* Tagged Templates
+* String Aradeğerleme
+* Çok Satırlı Stringler
+* Biçimlendirilmiş Şablonlar
 
-#### String Interpolation
-Another common use case is when you want to generate some string out of some static strings + some variables. For this you would need some *templating logic* and this is where *template strings* get their name from. Here's how you would potentially generate an html string previously:
+#### String Aradeğerleme
+Verilen string ile değişkenlerin değerini barındıran bir string çıktısı elde edilmek istendigi durumda yaygın olarak kullanılan bir yöntemdir. Bunun gerçekleştirilmesi için bir *şablonlama mantığı* gereksinimi bulundugundan *şablon stringler* adını buradan alır.
+Sıradan yollarla bir html string tanımlamak istendiğinde büyük olasılıkla şu şekilde gerçekleştirilecektir:
 
 ```ts
 var lyrics = 'Never gonna give you up';
 var html = '<div>' + lyrics + '</div>';
 ```
-Now with template strings you can just do:
+Şablon stringler ile bu şekilde yapılabilir:
 
 ```ts
 var lyrics = 'Never gonna give you up';
 var html = `<div>${lyrics}</div>`;
 ```
 
-Note that any placeholder inside the interpolation (`${` and `}`) is treated as a JavaScript expression and evaluated as such e.g. you can do fancy math.
+Aradeğerleme işlemi (`${` ve `}`)  içerisindeki yertutucuların birer JavaScript ifadesi olarak işlenip değerlendirileceğini unutmayınız. Örneğin burada bir takım matematiksel işlemler de gerçekleştirilebilir.
 
 ```ts
-console.log(`1 and 1 make ${1 + 1}`);
+console.log(`1 + 1 = ${1 + 1}`);
 ```
 
-#### Multiline Strings
-Ever wanted to put a newline in a JavaScript string? Perhaps you wanted to embed some lyrics? You would have needed to *escape the literal newline* using our favorite escape character `\`, and then put a new line into the string manually `\n` at the next line. This is shown below:
+#### Çok Satırlı Stringler
+Hiç JavaScript stringlerinde satırbaşı yapmak istediniz mi? Ya da bir şarkı sözü belirtmek istemediniz mi? Bu durumda stringimizin ilgili yerine favori *kaçış karakteri*miz olan `\` ile *satırbaşı karakteri* olan `\n`'yi 
+aşağıdaki örnekte gösterildiği gibi yerleştirmemiz gerekecektir:
 
 ```ts
 var lyrics = "Never gonna give you up \
 \nNever gonna let you down";
 ```
 
-With TypeScript you can just use a template string:
+TypeScript'teki şablon stringler aşağıdaki kolaylığı sağlamakta:
 
 ```ts
 var lyrics = `Never gonna give you up
 Never gonna let you down`;
 ```
 
-#### Tagged Templates
+#### Biçimlendirilmiş Şablonlar
+Şablon string tanımının önüne (`tag` olarak isimlendirilen) fonksiyonları ekleyebilir ve ilgili şablon stringinin ön işlenmesini gerçekleştirerek buradan dönen sonucu kullanabilirsiniz.
```

Comment:
String yabancı bir ifade olduğu için, sonrasında gelen ekleri apostrof ile ayırabiliriz. "String'inin" gibi.